File: q0179/q0179.java
import java.util.*;
import java.util.Comparator;

class Solution {
    private class MyComparator implements Comparator<String> {
        @Override
        public int compare(String x, String y) {
            return (y + x).compareTo(x + y);
        }
    }

    public String largestNumber(int[] nums) {
        List<String> resList = new ArrayList<>();
        for (int num : nums) {
            resList.add(String.valueOf(num));
        }
        Collections.sort(resList, new MyComparator());
        String res = new String();
        for (String s : resList) {
            res += s;
        }
        if (res.charAt(0) == '0')
            return "0";
        return res;
    }
}

File: q0179/q0179.py
class ComparatorKey(str):
    def __lt__(x, y):
        return x + y > y + x


class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        resArray = []
        for num in nums:
            resArray.append(str(num))
        resArray = sorted(resArray, key=ComparatorKey)
        res = "".join(resArray)
        return "0" if res[0] == "0" else res
